### ISSUE TITLE
Could you please provide a demo code for how to visualize the effect of mask decoder, such as the following figure?

### ISSUE THREAD

**Issue by littletomatodonkey** [facebookresearch/ConvNeXt-V2#42]
as the title, thanks!

![image](https://user-images.githubusercontent.com/14270174/225312291-414643e4-7925-4ae6-a592-77ada5830b6f.png)

I can just generate the following figure using the MAE code, and can not find out where to fix for ConvNext-V2.

![image](https://user-images.githubusercontent.com/14270174/225312598-53826f7c-7b12-4284-b1d7-01cca153eea7.png)

**Comment by Mogul5306** [comment]
Have you sovled the noise in the reconstructed images?

**Comment by hughsando** [comment]
There is nothing the the pipeline to force the output for the unmasked (ie, kept) patches to resemble the input because the differences are explicitly masked off in the `forward_loss` function.  All the masked patches have a 'mask_token' vector from the encoder, and none of the unmasked ones do.  I think it is the relative magnitudes of the mask_token and the 'pred' coefficients that cause high values in the reconstruction.
You can factor in a little of the unmasked pixels in the loss if the reconstruction is important to you - I am not sure how this affects performance on downstream tasks.

```python
def forward_loss(self, imgs, pred, mask):
    ...
    #loss = (loss * mask).sum() / mask.sum()

    # Factor in a little of the unmasked stuff too:
    loss = 0.1*loss.mean()  + 0.9* (loss * mask).sum() / mask.sum()

```

Some experiments with scaling the 'pred' vector also seem to help - although I am not using sparse tensors, so I may have made a mistake elsewhere.
```python
    torch.nn.init.xavier_uniform_(self.pred.weight,0.01)
    torch.nn.init.normal_(self.mask_token, std=10.0)
```
